include/deal.II/grid/manifold_lib.h

```diff
@@ -593,12 +593,16 @@ class FunctionManifold : public ChartManifold<dim, spacedim, chartdim>
    *
    * The tolerance argument is used in debug mode to actually check that the
    * two functions are one the inverse of the other.
+   *
+   * By default, only pointers to the passed functions are stored.
+   * Use @p owns_functions to change this behaviour.
```

Comment:
I would prefer to make it clearer that you are passing memory ownership by introducing a
```
 FunctionManifold(
    std::unique_ptr<Function<chartdim>> push_forward_function,    
    std::unique_ptr<Function<spacedim>> pull_back_function,
    ...
```
constructor.

Reply:
Hm, you mean by invoking std::move(push_forward_function) in the constructor and assigning the result to a member variable? BTW, this will require changing the member type from SmartPointer to std::unique_ptr, won't it?

Reply:
> Hm, you mean by invoking std::move(push_forward_function) in the constructor and assigning the result to a member variable?

Yes.


> BTW, this will require changing the member type from SmartPointer to std::unique_ptr, won't it?

Not necessarily. We can do internally whatever we want, e.g. obtaining the underlying pointer via `std::unique_ptr::release()`.